City-of-Helsinki/drupal-module-helfi-navigation
# UHF-9824: Allow relative links

## How to install

### Etusivu
- Make sure Etusivu instance is up and running
- Change the UID 1 user's password  to `123` on Etusivu: `drush upwd helfi-admin 123` for example

### Some other instance
- `composer require drupal/helfi_navigation:dev-UHF-9824`
- `composer require drupal/hdbt:dev-UHF-9824`
- Add navigation API keys to your `local.settings.php`:
  ```php
    $config['helfi_api_base.api_accounts']['vault'][] = $pubsub_vault;
    $config['helfi_api_base.api_accounts']['vault'][] = [
      'id' => 'helfi_navigation',
      'plugin' => 'authorization_token',
      'data' => base64_encode('helfi-admin:123'), // 123 is the password of helfi-admin account
    ];
  ```

## How to test
- Run `drush php-eval "_helfi_navigation_queue_item('main', 'en', 'update');" && drush cron && drush cr`
- Make sure local URLs are synced with domain (`https://helfi-etusivu.docker.so/en/api/v1/global-menu/liikenne`)
- Add these to your `sites/default/local.services.yml` file: 
   ```yaml 
    parameters:
      helfi_navigation.absolute_url_always: false
   ```
- Re-run the `php-eval` command
- Make sure the synced URLs are relative (`https://helfi-etusivu.docker.so/en/api/v1/global-menu/liikenne`
- Test Global navigation blocks on desktop size
- Test Global navigation blocks on mobile size

## Other PRs
- https://github.com/City-of-Helsinki/drupal-hdbt/pull/926
